abdiff/helpers/timdex_sources.py

```diff
@@ -0,0 +1,93 @@
+"""abdiff.helpers.timdex_sources"""
+
+import datetime
+import logging
+import re
+
+import boto3  # type: ignore[import-untyped]
+
+from abdiff.config import Config
+
+logger = logging.getLogger(__name__)
+
+CONFIG = Config()
+
+
+def get_extracted_files_for_source(
+    source: str,
+    bucket: str = CONFIG.TIMDEX_BUCKET,
+) -> list[str]:
+    """List S3 URIs for extract files in TIMDEX S3 bucket for a given source."""
+    s3_client = boto3.client("s3")
+    files = []
+
+    paginator = s3_client.get_paginator("list_objects_v2")
+    page_iterator = paginator.paginate(Bucket=bucket, Prefix=source)
+
+    for page in page_iterator:
+        if "Contents" in page:
+            for obj in page["Contents"]:
+                if not obj["Key"].endswith("/"):  # skip folders
+                    s3_uri = f"s3://{bucket}/{obj['Key']}"
+                    files.append(s3_uri)  # noqa: PERF401
+
+    # filter where "extracted" in filename
+    return [file for file in files if "extracted" in file]
+
+
+def get_ordered_extracted_files_since_last_full_run(source: str) -> list[str]:
```

Comment:
Hmm, what do you think about: updating `utils.parse_timdex_filename` to accept an _optional_ second argument: a list of strings denoting filename detail components (e.g. "date", "run-type"); if the second arg is provided the function returns only the specified components? You could then update the "private" methods `is_full_extract`, `is_daily_extract`, and `_extract_date` to call the util. 🤔

Reply:
Interesting... 

While conceptually I like it, I think I might lean into this code being "extra" in the sense it shouldn't apply too much pressure for change to the "core" code.  If we had reason to extract single values from the filenames in "core" code, that'd be one thing... but I think we primarily want at least 3-4 items each time we use it.

Thoughts?

Reply:
Ah, I think it's fine as-is! I think considering the `parse_timdex_filename` utility as part of the "core" set of functions makes sense, too.